src/Console/DiffCommand.php

```diff
@@ -30,83 +25,21 @@ class DiffCommand extends Command
-        $connection    = $configuration->getConnection();
+        DependencyFactoryProvider               $provider,
+        ConfigurationFactory                    $configurationFactory
+    ): int {
+        $dependencyFactory = $provider->fromConnectionName($this->option('connection'));
+        $migrationConfig = $configurationFactory->getConfigAsRepository($this->option('connection'));
 
-        // Overrule the filter
-        if ($filterExpr = $this->option('filter-expression')) {
-            $connection->getConfiguration()->setFilterSchemaAssetsExpression($filterExpr);
-        }
-
-        $fromSchema = $connection->getSchemaManager()->createSchema();
-        $toSchema   = $this->getSchemaProvider($em)->createSchema();
-
-        // Drop tables which don't suffice to the filter regex
-        if ($filterExpr = $connection->getConfiguration()->getFilterSchemaAssetsExpression()) {
-            foreach ($toSchema->getTables() as $table) {
-                $tableName = $table->getName();
-                if (!preg_match($filterExpr, $this->resolveTableName($tableName))) {
-                    $toSchema->dropTable($tableName);
-                }
-            }
-        }
-
-        $up   = $builder->up($configuration, $fromSchema, $toSchema);
-        $down = $builder->down($configuration, $fromSchema, $toSchema);
+        $command = new \Doctrine\Migrations\Tools\Console\Command\DiffCommand($dependencyFactory);
 
-        if (!$up && !$down) {
-            $this->error('No changes detected in your mapping information.');
-            return;
+        if ($this->input->getOption('filter-expression') === null) {
+            $this->input->setOption('filter-expression', $migrationConfig->get('schema.filter'));
         }
 
-        $path = $generator->generate(
-            $configuration,
-            false,
-            false,
-            $up,
-            $down
-        );
-
-        $this->line(sprintf('Generated new migration class to "<info>%s</info>" from schema differences.', $path));
-    }
-
-    /**
-     * @param EntityManagerInterface $em
-     *
-     * @return OrmSchemaProvider
-     */
-    protected function getSchemaProvider(EntityManagerInterface $em): OrmSchemaProvider
-    {
-        return new OrmSchemaProvider($em);
-    }
-
-    /**
-     * Resolve a table name from its fully qualified name. The `$name` argument
-     * comes from Doctrine\DBAL\Schema\Table#getName which can sometimes return
-     * a namespaced name with the form `{namespace}.{tableName}`. This extracts
-     * the table name from that.
-     *
-     * @param string $name
-     *
-     * @return string
-     */
-    protected function resolveTableName(string $name): string
-    {
-        $pos = strpos($name, '.');
-
-        return false === $pos ? $name : substr($name, $pos + 1);
+        return $command->run($this->getDoctrineInput($command), $this->output->getOutput());
```

Comment:
Before (L6, LD/orm 1.5, LD/mig 2.3.1)
```
$ php artisan doctrine:migrations:diff                        
No changes detected in your mapping information.
```

After (L9, LD/orm 1.9, LD/mig this)
```
$ php artisan doctrine:migrations:diff                 

   Doctrine\Migrations\Generator\Exception\NoChangesDetected 

  No changes detected in your mapping information.

  at vendor/doctrine/migrations/lib/Doctrine/Migrations/Generator/Exception/NoChangesDetected.php:13
      9▕ final class NoChangesDetected extends RuntimeException implements GeneratorException
     10▕ {
     11▕     public static function new(): self
     12▕     {
  ➜  13▕         return new self('No changes detected in your mapping information.');
     14▕     }
     15▕ }
     16▕ 

      +3 vendor frames 
  4   /Users/matt/Dropbox/HackspaceInstrumentation/github/laravel-doctrine/migrations/src/Console/DiffCommand.php:43
      Symfony\Component\Console\Command\Command::run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\ConsoleOutput))

      +13 vendor frames 
  18  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

i think we should catch the error, this will return to old behaviour
```suggestion
                try {
            return $command->run($this->getDoctrineInput($command), $this->output->getOutput());
        } catch (NoChangesDetected $exception) {
            $this->error($exception->getMessage());

            return 0;
        }
```

will need `use Doctrine\Migrations\Generator\Exception\NoChangesDetected;`

Reply:
Fixed !